q0Utils.py

```diff
@@ -0,0 +1,779 @@
+# refilling the cryomodule we refill to at least 95%.
+MIN_DS_LL = 90
+MAX_DS_LL = 95
+
+MIN_US_LL = 66
+
+# Used to reject data where the JT valve wasn't at the correct position
+VALVE_POS_TOL = 2
+
+# Used to reject data where the cavity heater wasn't at the correct value
+HEATER_TOL = 1.2
+
+# The minimum acceptable run length is ten minutes (600 seconds)
+MIN_RUN_DURATION = 200
+
+# We want the liquid level to drop by at least 2.5% during our runs. This isn't
+# actually enforced however, unlike the run duration.
+TARGET_LL_DIFF = 4
+
+# Used to reject data where the cavity amplitude wasn't at the correct value
+AMPLITUDE_TOL = 0.3
+
+# We fetch data from the JLab archiver with a program called MySampler, which
+# samples the chosen PVs at a user-specified time interval. Increase to improve
+# statistics, decrease to lower the size of the CSV files and speed up
+# MySampler data acquisition.
+ARCHIVER_TIME_INTERVAL = 1
+
+# Used in custom input functions
+ERROR_MESSAGE = "Please provide valid input"
+
+# This is used to suppress the output of the caput function.
+FNULL = open(devnull, "w")
+
+# The starting point for our calibration
+INITIAL_CAL_HEAT_LOAD = 8
+
+# The number of distinct heater settings we're using for cryomodule calibrations
+NUM_CAL_STEPS = 7
+
+NUM_LL_POINTS_TO_AVG = 25
+
+CAV_HEATER_RUN_LOAD = 24
+FULL_MODULE_CALIBRATION_LOAD = 80
+
+CAL_HEATER_DELTA = 8
+
+JT_SEARCH_TIME_RANGE: timedelta = timedelta(hours=24)
+JT_SEARCH_OVERLAP_DELTA: timedelta = timedelta(minutes=30)
+DELTA_NEEDED_FOR_FLATNESS: timedelta = timedelta(hours=2)
+
+FULL_CALIBRATION_FILENAME_TEMPLATE = "calibrationsCM{CM}.json"
+CAVITY_CALIBRATION_FILENAME_TEMPLATE = "cav{CAV}/calibrationsCM{CM}CAV{CAV}.json"
+
+RUN_STATUS_MSSG = ("\nWaiting for the LL to drop {DIFF}% "
+                   "or below {MIN}%...".format(MIN=MIN_DS_LL, DIFF=TARGET_LL_DIFF))
+
+SSA_SLOPE_CHANGE_TOL = 0.15
+LOADED_Q_CHANGE_TOL = 0.15e7
+CAVITY_SCALE_CHANGE_TOL = 0.2
+
+# Swapnil requests 1%/min
+JT_STEP_SIZE_PER_SECOND = 1 / 60
+
+JT_MANUAL_MODE_VALUE = 0
+JT_AUTO_MODE_VALUE = 1
+
+RF_MODE_SELAP = 0
+RF_MODE_SELA = 1
+RF_MODE_SEL = 2
+RF_MODE_SEL_RAW = 3
+RF_MODE_PULSE = 4
+RF_MODE_CHIRP = 5
+
+SAFE_PULSED_DRIVE_LEVEL = 15
+
+ARCHIVER = Archiver("lcls")
+
+
+# The calculated Q0 value for this run. Formula from Mike Drury
+# (drury@jlab.org) to calculate Q0 from the measured heat load on a cavity,
+# the RF gradient used during the test, and the pressure of the incoming
+# 2 K helium.
+def calcQ0(amplitude: float, rfHeatLoad: float, avgPressure: float):
+    # The initial Q0 calculation doesn't account for the temperature
+    # variation of the 2 K helium
+    rUponQ = 1012
+    
+    uncorrectedQ0 = (((amplitude * 1000000) ** 2)
+                     / (rUponQ * rfHeatLoad))
+    
+    # uncorrectedQ0 = ((grad * 1000000) ** 2) / (939.3 * rfHeatLoad)
+    
+    # We can correct Q0 for the helium temperature!
+    tempFromPress = (avgPressure * 0.0125) + 1.705
```

Comment:
Technically, we shouldn't need this since we are operating at 2K. That being said, fine to leave it in. (Stupid question: Have you double checked that the correction is actually zero for 2K?)

Reply:
The correction is non zero, but I only get Dan's numbers with the correction (?!)